File: backend/src/main/java/io/github/pudo58/base/service/OrderService.java
package io.github.pudo58.base.service;

import io.github.pudo58.base.entity.Order;
import io.github.pudo58.dto.CommonRequest;
import io.github.pudo58.dto.OrderActionRequest;
import io.github.pudo58.dto.OrderRequest;
import jakarta.mail.MessagingException;
import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;

import java.io.IOException;
import java.net.URISyntaxException;
import java.security.NoSuchAlgorithmException;
import java.util.UUID;

public interface OrderService {
    void saveContact(OrderRequest model);
    ResponseEntity<?> createOrder(OrderRequest model) throws NoSuchAlgorithmException, IOException, InterruptedException, URISyntaxException;

    ResponseEntity<?> getOrderInfo(OrderRequest model) throws URISyntaxException, NoSuchAlgorithmException, IOException, InterruptedException;

    ResponseEntity<?> getQrCode(UUID orderId) throws NoSuchAlgorithmException, IOException, InterruptedException, URISyntaxException;

    ResponseEntity<?> cancelOrder(OrderRequest model);

    ResponseEntity<?> getOrder(UUID orderId);

    ResponseEntity<?> approveOrder(UUID orderId);

    ResponseEntity<?> getDefaultContact();

//    for admin
    Page<Order> findBySearch(CommonRequest model);

    ResponseEntity<?> approve(OrderActionRequest model) throws MessagingException, IOException;

    ResponseEntity<?> reject(OrderActionRequest model);

    ResponseEntity<?> shipping(OrderActionRequest model);

    ResponseEntity<?> receivedOrder(OrderActionRequest model);

    // for user
    Page<Order> findByUser(CommonRequest model);
}

File: frontend/src/base/service/order.service.ts
import axios from "axios";

export class OrderService {
    public url = '/v2/order';
    public adminUrl = '/order';
    private store: any;

    constructor(store: any) {
        this.store = store;
    }

    async createOrder(model: any) {
        const response = await axios.post(`${this.url}/create`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async getDefaultContact() {
        const response = await axios.get(`${this.url}/getDefaultContact`, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async getOrderInfo(model: any) {
        const response = await axios.post(`${this.url}/getOrderInfo`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async getQrCode(orderId: any) {
        const response = await axios.get(`${this.url}/getQrCode/${orderId}`, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async cancelOrder(model: any) {
        const response = await axios.post(`${this.url}/cancel`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async findByUser(model: any) {
        const response = await axios.post(`${this.url}/findByUser`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async receive(model: any) {
        const response = await axios.post(`${this.url}/received`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async getOrder(orderId: any) {
        const response = await axios.get(`${this.url}/get/${orderId}`, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

//    for admin
    async findBySearch(model: any) {
        const response = await axios.post(`${this.adminUrl}/findBySearch`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async approve(model: any) {
        const response = await axios.post(`${this.adminUrl}/approve`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async reject(model: any) {
        const response = await axios.post(`${this.adminUrl}/reject`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async shipping(model: any) {
        const response = await axios.post(`${this.adminUrl}/shipping`, model, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }

    async get(id: string) {
        const response = await axios.get(`${this.adminUrl}/get/${id}`, {
            headers: {
                Authorization: `${this.store?.getters.getToken}`
            }
        });
        return response.data;
    }
}